vcr/oidc4vci/validators.go

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (C) 2023 Nuts community
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+package oidc4vci
+
+import (
+	"errors"
+	ssi "github.com/nuts-foundation/go-did"
+	"github.com/nuts-foundation/go-did/vc"
+)
+
+// ValidateCredentialDefinition validates the CredentialDefinition according to the VerifiableCredentialJSONLDFormat format
+func ValidateCredentialDefinition(definition *CredentialDefinition, isOffer bool) error {
```

Comment:
doesn't that just move the `nil` check outside this function before calling the validator on the `CredentialDefinition`?

Reply:
I've made it a function on the `CredentialDefinition`, but it still contains the nil check.

Reply:
That's because of the pointer receiver. Do we always need a definition on an offer. Then the offer can also remove the pointer from the definition. You'll get an empty struct but validation would fail for that as well. Less pointers equals easier life.

Reply:
the `CredentialDefinition` is optional since it is only part of the verifiable credential flows, not de mDL flows. We only support `ldp_vc` for now where it is required.

Reply:
Unless you don't care about the specific error and just let it fail on missing @context. I prefer the current solution as it is closest to the spec and therefor easiest to understand/extend.